### PR TITLE
Fix TODOs and some minor improvements

### PR DESCRIPTION
**Pull request by alexbruy**
Add check for geometry validity which works also for multipart lines/polygons
Display "Done" button when editing multipoint layers

### CODE REVIEW

File: app/maptools/recordingmaptool.cpp [hunk 1]
```diff
@@ -336,11 +349,41 @@ bool RecordingMapTool::hasValidGeometry() const
...
+          if ( ( *it )->nCoordinates() < 3 )
+          {
+            return false;
+          }
+        }
+        return true;
+      }
+      else
+      {
+        return mRecordedGeometry.constGet()->nCoordinates() >= 3;
```
Comment (tomasMizera) [inline app/maptools/recordingmaptool.cpp:385]: Shouldn't it actually be 4? (When ring is closed, with three points it is 4 in total - first duplicated)

```suggestion
        return mRecordedGeometry.constGet()->nCoordinates() >= 4;
```

File: app/maptools/recordingmaptool.cpp [hunk 2]
```diff
@@ -336,11 +349,41 @@ bool RecordingMapTool::hasValidGeometry() const
...
     }
     else if ( mLayer->geometryType() == QgsWkbTypes::PolygonGeometry )
     {
-      return mRecordedGeometry.constGet()->nCoordinates() >= 3;
+      if ( mRecordedGeometry.isMultipart() )
+      {
+        const QgsAbstractGeometry *geom = mRecordedGeometry.constGet();
+        for ( auto it = geom->const_parts_begin(); it != geom->const_parts_end(); ++it )
+        {
+          if ( ( *it )->nCoordinates() < 3 )
```
Comment (tomasMizera) [inline app/maptools/recordingmaptool.cpp:376]: Same, 4 instead of 3

`nCoordinates < 4`

File: app/inpututils.cpp [hunk 3]
```diff
@@ -1088,6 +1088,11 @@ bool InputUtils::isPointLayer( QgsVectorLayer *layer )
   return geometryFromLayer( layer ) == "point";
 }
 
+bool InputUtils::isMultiPointLayer( QgsVectorLayer *layer )
```
Comment (tomasMizera) [inline app/inpututils.cpp:1091]: maybe better to have more generic `isMultiPart()` function

Comment (tomasMizera) [reply app/inpututils.cpp:1091]: then in QML we can use `__inputUtils.isPointLayer( __activeLayer.vectorLayer ) && !__inputUtils.isMultiPartLayer( __activeLayer.vectorLayer )`

**Review by tomasMizera — APPROVED** [review]

**Comment by tomasMizera** [comment]
We have some issues with autotests here ..

**Comment by tomasMizera** [comment]
can you please check why is that @alexbruy ?

**Comment by alexbruy** [comment]
No idea why is that. Failed test is not related to the any of the changes in this PR and passes locally.

**Comment by alexbruy** [comment]
@tomasMizera should be fixed now.

**Review by tomasMizera — APPROVED** [review]

**Pull request merged**